### PR TITLE
[BottomDrawer] Allow touch events to propagate to delegate for  MDCBottomNavigationDrawer

### PR DESCRIPTION
**Pull request by randallli**
[BottomDrawer] Allow touch events to propagate to delegate for  MDCBottomNavigationDrawer

Allow the drawer to optionally forward touch events to the delegate for handling. This enables tap thru to the underlying VC if the client needs that behavior.

An example use case would be when the client wants to present the drawer on top of a VC that has controls that they want to still be tappable. For example: A video player or a podcast player with play/pause controls. Then the client VC could still receive the tap event on the control and respond to that and close the drawer at the same time. This would allow the user to save a tap.

this is a second attempt at #8486

original cl/270318683

### CODE REVIEW

**Comment by jverkoey** [comment]
Is there an associated issue for this PR?

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h [hunk 1]
```diff
@@ -102,6 +102,24 @@
  */
 @property(nonatomic, assign) CGFloat maximumInitialDrawerHeight;
 
+/**
+ A flag allowing clients to opt-out of the drawer closing when the user taps outside the content.
+
+ @default YES The drawer should autohide on tap.
+ */
+@property(nonatomic, assign) BOOL shouldAutoDismissOnTap;
```
Comment (jverkoey) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h]: Consider modeling this API after `MDCDialogPresentationController`'s `dismissOnBackgroundTap`:

```objc

/**
 Should a tap on the dimmed background view dismiss the presented controller.

 Defaults to YES.
 */
@property(nonatomic, assign) BOOL dismissOnBackgroundTap;
```

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h [hunk 2]
```diff
@@ -102,6 +102,24 @@
...
+
+/**
+ A flag allowing clients to opt-in to handling touch events.
+
+ @default NO The drawer will not forward touch events.
```
Comment (jverkoey) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h:179]: Please answer the following questions in the docs here:

- What is a touch event in this context?
- Where are the touch events being forward to?

**Review by jverkoey — CHANGES_REQUESTED** [review]

**Comment by randallli** [comment]
> Is there an associated issue for this PR?

no, this is a internal contribution.

**Comment by jverkoey** [comment]
> no, this is a internal contribution.

Please link to the CL # in the PR description then.

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h (hunk 1 above)
Comment (randallli) [reply components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h]: ah good idea.

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h (hunk 2 above)
Comment (randallli) [reply components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h:179]: Im not sure if I understand. Do you mean that the "discussion" below is not sufficient.

Comment (jverkoey) [reply components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h:179]: Yes, I’m not sure what the delegate means in this context. Is a specific selector invoked?

Comment (jverkoey) [reply components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h]: Yes, I’m not sure what the delegate means in this context. Is a specific selector invoked?

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h [hunk 3]
```diff
@@ -102,6 +102,24 @@
...
+
+ @discussion If set to YES and the delegate is an instance of @UIResponder, then the touch events
+ will be forwarded along to the delegate. Note: @shouldAutoDismissOnTap should also be set to NO
+ so that the events will propagate properly.
```
Comment (jverkoey) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h]: What happens if shouldAutoDismissOnTap is Yes? Should this configuration throw an assertion?

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h (hunk 2 above)
Comment (randallli) [reply components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h:179]: more information added.

**Review by randallli — COMMENTED** [review]
> What happens if shouldAutoDismissOnTap is Yes? Should this configuration throw an assertion?

If dismissOnBackgroundTap is YES then this method will never be called since the TapGesture will eat the touch event and there will be nothing left to process. -hiediutley

**Review by jverkoey — APPROVED** [review]

**Pull request merged**